src/common/tests/MoveStringTest.cpp

```diff
@@ -0,0 +1,146 @@
+#include "boost/test/unit_test.hpp"
```

Comment:
There is class string test src/common/classes/misc/string_test.cpp. It's very old (like string itself) and definitely does not use boost-related features, but there are checks for most of existing functions in the class. May be instead adding separate test it's better to have single  one, sooner of all using boost. 

PS. I think keeping performance comparison with std::basic_string does not make sense currently and may be cleaned out. First of all because our string is almost never used in performance-critical paths of execution.

Reply:
I have moved the old test to the Boost Test Library. Some tests for the comparison method do not pass.
```
PathName c = "Aa";
PathName d = "AB";
check(c.compare(d), -1);
```
`StringTest.cpp(530): error: in "StringSuite/StringFunctionalTests/CompareTests1": check c.compare(d) == -1 has failed [7967 != -1]`
 
The result is expected to be 0, -1 or 1, but in reality some positive value is obtained -just like, the result of the memcmp function.

Reply:
May be use something like:
   check(sign(c.compare(d)), -1);

Reply:
Ok, but originally I wanted to know if the compare method returns the value correctly. From your answer I understand that yes, the problem is in too old test. I will fix it